dubbo-registry/dubbo-registry-zookeeper/src/main/java/org/apache/dubbo/registry/zookeeper/ZookeeperRegistry.java

```diff
@@ -133,21 +130,18 @@ public void doSubscribe(final URL url, final NotifyListener listener) {
                 String root = toRootPath();
                 ConcurrentMap<NotifyListener, ChildListener> listeners = zkListeners.get(url);
                 if (listeners == null) {
-                    zkListeners.putIfAbsent(url, new ConcurrentHashMap<NotifyListener, ChildListener>());
+                    zkListeners.putIfAbsent(url, new ConcurrentHashMap<>());

```

Comment:
I think the whole can be replace by in line no 133 with below code 
`ConcurrentMap<NotifyListener, ChildListener> listeners = zkListeners.computeIfAbsent(url,k-> new ConcurrentHashMap<>());`

what do you say?

Reply:
Sure.